src/xrpld/app/misc/CredentialHelpers.cpp

```diff
@@ -225,6 +226,50 @@ makeSorted(STArray const& in)
     return out;
 }
 
+NotTEC
+checkArray(STArray const& credentials, unsigned maxSize, beast::Journal j)
+{
+    if (credentials.empty() || (credentials.size() > maxSize))
+    {
+        JLOG(j.trace()) << "Malformed transaction: "
+                           "Invalid credentials size: "
+                        << credentials.size();
+        return temMALFORMED;
+    }
+
+    std::unordered_set<uint256> duplicates;
+    for (auto const& credential : credentials)
+    {
+        auto const& issuer = credential[sfIssuer];
+        if (!issuer)
+        {
+            JLOG(j.trace()) << "Malformed transaction: "
+                               "Issuer account is invalid: "
+                            << to_string(issuer);
+            return temINVALID_ACCOUNT_ID;
+        }
+
+        auto const ct = credential[sfCredentialType];
+        if (ct.empty() || (ct.size() > maxCredentialTypeLength))
+        {
+            JLOG(j.trace()) << "Malformed transaction: "
+                               "Invalid credentialType size: "
+                            << ct.size();
+            return temMALFORMED;
```

Comment:
Ed recommends `temINVALID` over `temINVALID` (though it doesn't make too much of a difference)

Reply:
changed to temARRAY_EMPTY

Reply:
It's not an array.